python/ray/_raylet.pyx

```diff
@@ -304,8 +304,8 @@ cdef prepare_args(
         else:
             serialized_arg = worker.get_serialization_context().serialize(arg)
             metadata = serialized_arg.metadata
-            metadata_fields = metadata.split(b",")
             if language != Language.PYTHON:
+                metadata_fields = metadata.split(b",")
```

Comment:
a tiny perf improvement on the critical path. `metadata_fields` is only used in this (almost never hit) branch